satellite/audit/service.go

```diff
@@ -41,20 +50,32 @@ type Service struct {
 	Reporter reporter
 
 	Loop sync2.Cycle
+
+	// for audit 2.0 using metainfoloop
+	ReservoirLoop   sync2.Cycle
+	overlay         *overlay.Service
+	MetainfoLoop    *metainfo.Loop
+	reservoirConfig ReservoirConfig
 }
 
 // NewService instantiates a Service with access to a Cursor and Verifier
 func NewService(log *zap.Logger, config Config, metainfo *metainfo.Service,
 	orders *orders.Service, transport transport.Client, overlay *overlay.Service,
-	containment Containment, identity *identity.FullIdentity) (*Service, error) {
+	containment Containment, identity *identity.FullIdentity, loop *metainfo.Loop) (*Service, error) {
 	return &Service{
 		log: log,
 
 		Cursor:   NewCursor(metainfo),
 		Verifier: NewVerifier(log.Named("audit:verifier"), metainfo, transport, overlay, containment, orders, identity, config.MinBytesPerSecond, config.MinDownloadTimeout),
 		Reporter: NewReporter(log.Named("audit:reporter"), overlay, containment, config.MaxRetriesStatDB, int32(config.MaxReverifyCount)),
 
-		Loop: *sync2.NewCycle(config.Interval),
+		// for audit 2.0
+		overlay:         overlay,
+		reservoirConfig: config.ReservoirConfig,
+		MetainfoLoop:    loop,
+
+		Loop:          *sync2.NewCycle(config.Interval),
```

Comment:
This is the old "per segment" auditing loop, right? We are going to remove it in favor of the new reservoir loop, right?

Reply:
yes, probably not in this pr though